### ISSUE TITLE
Option to swap out VSCode for VSCodium

### ISSUE THREAD

**Issue by TiemenSch** [ublue-os/bluefin#1691]
### Describe the package

I'd like to request the package `VSCodium` because I would like to keep the image a bit more FOSS and less of a "telemetry farm" for MS.

https://vscodium.com/

How would I go about replacing this? From what I can see it's quite heavily integrated in the system configuration and devcontainer setup. 

To begin with, I think it would default to installing vscodium (of course) and adding a command alias `code` that redirects to `codium`. 

The harder part is to check whether all extensions also exist on the open extension marketplace https://open-vsx.org and to migrate the IDE/extension config.

### Image

All Images

**Comment by castrojo** [comment]
You'd have to make a custom image for this: https://github.com/ublue-os/image-template

**Comment by TiemenSch** [comment]
I'm looking for a solution where it's all bluefin and it's tweaks, with this as the key change. I was wondering whether it is a flavor uBlue would consider.

Then again, I might just start cherry picking all changes in this repo to be built using that template in the end anyways.

**Comment by m2Giles** [comment]
I would recommend going the custom image route if you want to completely remove and replace it. At this time this is not something we are considering for Bluefin given the capability gap between the two products.

Additionally, you do not have to use VSCode while using Bluefin. If having it appear in your application list is something you want to avoid, you can copy the desktop file to a writable data-dirs directory and add the hidden key to the desktop file.

**Comment by salim-b** [comment]
> (...) given the capability gap between the two products.

Would you care to further elaborate on what capabilities VSCodium is missing exactly (besides obtaining extension from the [Open VSX Registry](https://open-vsx.org/) instead of Microsofts proprietary Marketplace)? I'm sincerely interested.

**Comment by TiemenSch** [comment]
The editor itself is pretty much identical, but VSCodium skips the telemetry parts wherever it can. This results in having to default to the open VSX registry for extensions.

I personally don't use the containerized development aspects of uBlue that much, but since that is promoted as a big part of the DX I tried to see some feature parity was possible. Most notable differences I came across:

- Configure PyRight to fill in most of the important bits of the Microsoft Python extension that are missing.
- Look for alternatives and (default) configurations for remote/containerized development. The default container/SSH was missing I believe. 

I might have overlooked some Microsoft-only extensions. Bundling an extension for VSCode or VSCodium is no different, so for most open-source extensions it's just adding a CI job to publish it to Open VSX as well. It's just MS that has an incentive to close some parts down for varying reasons such as AI training or usage and/or telemetry.
